### ISSUE TITLE
[BUG] Spark 3.2.0 test fails due to SPARK-34906 Refactor TreeNode's children handling methods into specialized traits

### ISSUE THREAD

**Issue by tgravescs** [NVIDIA/spark-rapids#2052]
**Describe the bug**
nightly spark 3.2.0 test fail with:
```
[2021-03-31T07:24:52.771Z] [32mHashAggregatesSuite:[0m
[2021-03-31T07:25:02.731Z] [31m*** RUN ABORTED ***[0m
[2021-03-31T07:25:02.731Z] [31m  java.lang.NoSuchMethodError: org.apache.spark.sql.execution.UnaryExecNode.children$(Lorg/apache/spark/sql/execution/UnaryExecNode;)Lscala/collection/Seq;[0m
[2021-03-31T07:25:02.731Z] [31m  at com.nvidia.spark.rapids.GpuHashAggregateExec.children(aggregate.scala:302)[0m
[2021-03-31T07:25:02.731Z] [31m  at org.apache.spark.sql.catalyst.trees.TreeNode.containsChild$lzycompute(TreeNode.scala:120)[0m
[2021-03-31T07:25:02.731Z] [31m  at org.apache.spark.sql.catalyst.trees.TreeNode.containsChild(TreeNode.scala:120)[0m
[2021-03-31T07:25:02.731Z] [31m  at org.apache.spark.sql.catalyst.trees.TreeNode.mapChildren(TreeNode.scala:356)[0m
[2021-03-31T07:25:02.731Z] [31m  at org.apache.spark.sql.catalyst.trees.TreeNode.transformUp(TreeNode.scala:337)[0m
[2021-03-31T07:25:02.731Z] [31m  at com.nvidia.spark.rapids.GpuOverrides.addSortsIfNeeded(GpuOverrides.scala:2992)[0m
[2021-03-31T07:25:02.731Z] [31m  at com.nvidia.spark.rapids.GpuOverrides.applyOverrides(GpuOverrides.scala:2950)[0m
[2021-03-31T07:25:02.731Z] [31m  at com.nvidia.spark.rapids.GpuOverrides.apply(GpuOverrides.scala:2911)[0m
[2021-03-31T07:25:02.731Z] [31m  at com.nvidia.spark.rapids.GpuOverrides.apply(GpuOverrides.scala:2897)[0m
[2021-03-31T07:25:02.731Z] [31m  at org.apache.spark.sql.execution.ApplyColumnarRulesAndInsertTransitions.$anonfun$apply$1(Columnar.scala:532)[0m
```

This looks like this is from:

https://issues.apache.org/jira/browse/SPARK-34906

https://github.com/apache/spark/commit/bd0990e3e813d17065c593fc74f383b494fe8146

**Comment by jlowe** [comment]
Note that #2056 is removing Spark 3.2 from some of the CI builds, so when this is fixed those builds should be reinstated.

**Comment by gerashegalov** [comment]
Fixed in #3223
